run/README.md

```diff
@@ -0,0 +1,90 @@
+# Google Cloud Run Java Samples
+
+[![Open in Cloud Shell][shell_img]][shell_link]
+
+[shell_img]: http://gstatic.com/cloudssh/images/open-btn.png
+[shell_link]: https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/java-docs-samples&page=editor&open_in_editor=blog/README.md
+
+This directory contains samples for [Google Cloud Run](https://cloud.run). [Cloud Run][run_docs] runs stateless [containers](https://cloud.google.com/containers/) on a fully managed environment or in your own GKE cluster.
+
+## Samples
+
+|           Sample                |        Description       |     Deploy    |
+| ------------------------------- | ------------------------ | ------------- |
+|[Hello World][helloworld]&nbsp;&#10149; | Quickstart | [<img src="https://storage.googleapis.com/cloudrun/button.svg" alt="Run on Google Cloud" height="30">][run_button_helloworld] |
+|[Cloud Pub/Sub][pubsub] | Handling Pub/Sub push messages | [<img src="https://storage.googleapis.com/cloudrun/button.svg" alt="Run on Google Cloud" height="30">][run_button_pubsub] |
+
+For more Cloud Run samples beyond Java, see the main list in the [Cloud Run Samples repository](https://github.com/GoogleCloudPlatform/cloud-run-samples).
+
+## Setup
+
+1. [Set up for Cloud Run development](https://cloud.google.com/run/docs/setup)
+
+2. Clone this repository:
+
+    ```
+    git clone https://github.com/GoogleCloudPlatform/java-docs-samples.git
+    ```
+
+    Note: Some samples in the list above are hosted in other repositories. They are noted with the symbol "&#10149;".
+
+
+## How to run a sample locally
+
+1. [Install docker locally](https://docs.docker.com/install/)
+
+2. [Build the sample container](https://cloud.google.com/run/docs/building/containers#building_locally_and_pushing_using_docker):
+
+    ```
+    export SAMPLE=$sample
+    cd $SAMPLE
+    docker build --tag $sample .
+    ```
+
+3. [Run containers locally](https://cloud.google.com/run/docs/testing/local)
+
+    With the built container:
+
+    ```
+    PORT=8080 && docker run --rm -p 8080:${PORT} -e PORT=${PORT} $SAMPLE
+    ```
+
+    Overriding the built container with local code:
+
+    ```
+    PORT=8080 && docker run --rm \
+        -p 8080:${PORT} -e PORT=${PORT} \
+        -v $PWD:/app $SAMPLE
+    ```
+
+    Injecting your service account key:
+
+    ```
+    export SA_KEY_NAME=my-key-name-123
+    PORT=8080 && docker run --rm \
+        -p 8080:${PORT} -e PORT=${PORT} \
+        -e GOOGLE_APPLICATION_CREDENTIALS=/tmp/keys/${SA_KEY_NAME}.json \
+        -v $GOOGLE_APPLICATION_CREDENTIALS:/tmp/keys/${SA_KEY_NAME}.json:ro \
+        -v $PWD:/app $SAMPLE
+    ```
+
+## Deploying
+
+```
+gcloud builds submit --tag gcr.io/${GOOGLE_CLOUD_PROJECT}/${SAMPLE}
```

Comment:
This is probably the experience you want, but given that you built everything in the step above, you could just give a `docker push gcr.io/$PROJECT/$SAMPLE` and a gcloud run ...

Reply:
This is consistent with the other Cloud Run READMEs. I did it this way because authenticating docker with gcr takes an extra step, and Cloud Build sometimes has different results than a local build.

I think it's fine to change on a per language basis as long as we add the docker auth step. Maybe we should also include a link to a good jib tutorial?